js/snippetPreview.js

```diff
@@ -0,0 +1,162 @@
+    return this.formatKeyword( title );
+};
+
+/**
+ * formats the url for the snippet preview
+ * @returns formatted url
+ */
+YoastSEO_SnippetPreview.prototype.formatCite = function() {
+    var cite = this.refObj.inputs.url;
+    cite = this.refObj.pageAnalyzer.YoastSEO_preProcessor.stripAllTags( cite );
+    return this.formatKeywordUrl( cite );
+};
+
+/**
+ * formats the metatext for the snippet preview, if empty runs getMetaText
+ * @returns formatted metatext
+ */
+YoastSEO_SnippetPreview.prototype.formatMeta = function() {
+    var meta = this.refObj.inputs.meta;
+    if(meta === ""){
+        meta = this.getMetaText();
+    }
+    meta = this.refObj.pageAnalyzer.YoastSEO_preProcessor.stripAllTags( meta );
+    meta = meta.substring(0,analyzerConfig.maxMeta);
+    return this.formatKeyword( meta );
+};
+
+/**
+ * formats the metatext, based on the keyword to select a part of the text.
+ * If no keyword matches, takes the first 156chars (depending on the config)
+ * @returns metatext
+ */
+YoastSEO_SnippetPreview.prototype.getMetaText = function() {
+    var indexMatches = this.getIndexMatches();
+    var periodMatches = this.getPeriodMatches();
+    var metaText = this.refObj.inputs.textString.substring(0, analyzerConfig.maxMeta);
+    var curStart = 0;
+    if(indexMatches.length > 0) {
+        for (var j = 0; j < periodMatches.length; ) {
+            if (periodMatches[0] < indexMatches[0] ) {
+                curStart = periodMatches.shift();
+            } else {
+                if( curStart > 0 ){
+                    curStart += 2;
+                }
+                break;
+            }
+        }
+        metaText = this.refObj.inputs.textString.substring( curStart, curStart + analyzerConfig.maxMeta );
+    }
+    return metaText;
+};
+
+/**
+ * Builds an array with all indexes of the keyword
+ * @returns Array with matches
+ */
+YoastSEO_SnippetPreview.prototype.getIndexMatches = function() {
+    var indexMatches = [];
+    var match;
+    var i = 0;
+    while ( ( match = this.refObj.inputs.textString.indexOf( this.refObj.inputs.keyword, i ) ) > -1 ) {
+        indexMatches.push( match );
+        i = match + this.refObj.inputs.keyword.length;
+    }
+    return indexMatches;
+};
+
+/**
+ * Builds an array with indexes of all sentence ends (select on .)
+ * @returns array with sentences
+ */
+YoastSEO_SnippetPreview.prototype.getPeriodMatches = function() {
+    var periodMatches = [0];
+    var match;
+    var i = 0;
+    while( ( match = this.refObj.inputs.textString.indexOf( ".", i ) ) > -1 ){
+        periodMatches.push( match );
+        i = match + 1;
+    }
+    return periodMatches;
+};
+
+/**
+ * formats the keyword for use in the snippetPreview by adding <strong>-tags
+ * @param textString
+ * @returns textString
+ */
+YoastSEO_SnippetPreview.prototype.formatKeyword = function( textString ) {
+    var replacer = new RegExp( this.refObj.inputs.keyword, "ig" );
```

Comment:
Perhaps add inline documentation to explain the regex modifiers ('ig') you're setting, so the code is easier to read